src/core/utils/qfieldcloudutils.cpp

```diff
@@ -65,7 +65,7 @@ const QString QFieldCloudUtils::getProjectId( const QString &fileName )
   QString basePath = QFileInfo( baseDir.path() ).canonicalFilePath();
   QString cloudPath = QFileInfo( localCloudDirectory() ).canonicalFilePath();
 
-  if ( basePath.startsWith( cloudPath ) )
+  if ( !cloudPath.isEmpty() && basePath.startsWith( cloudPath ) )
```

Comment:
CC @suricactus does that look safe (seems unrelated to this feature)

Reply:
Yes, it is. Cannonical path retuns empty string on the first clear install on a new device, which causes the `if` to fail (`"".startsWith("path/here")`). However, if you download your first project, all works fine. That's why it was hard to catch it and it was there for ages. Thanks to @nirvn that he caught it and for the fix.